### PR TITLE
Fix heading levels on the category pages

### LINKED ISSUE #5942: [A11Y] Skipped heading levels on category page
**Describe the bug**

The heading levels are skipped on category pages.

**To Reproduce**
1. Go to a category that has subcategories and events
2. Use WAVE or Headings Map (or a screen reader) to inspect the heading structure
3. Observe that heading level 2 and 3 are skipped

**Expected behavior**

Expected a correct heading level tree with no skipped headings.

**Additional context**

- https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships
- https://talk.getindico.io/t/info-and-relationships-semantic-headings/3202

### PR DESCRIPTION
**Pull request by foxbunny**
- Add visually hidden H2 to category list and event list
- Promote event list H4 to H3

Fixes #5942

### CODE REVIEW

File: indico/web/client/styles/base/_utilities.scss [hunk 1]
```diff
@@ -77,3 +77,11 @@
 .ellipsis {
   @include ellipsis();
 }
+
+%visually-hidden {
```
Comment (tomasr8) [inline indico/web/client/styles/base/_utilities.scss:81]: Once merged, do we want to reuse the placeholder from the new design system? Though I see that's it not exactly the same..
Also, could you please rebase this PR? Thanks!

Comment (foxbunny) [reply indico/web/client/styles/base/_utilities.scss:81]: Yeah, once the design system is merged, I'm going to unify all these randomly added placeholders under it.

Comment (foxbunny) [reply indico/web/client/styles/base/_utilities.scss:81]: >Also, could you please rebase this PR? Thanks!

I'll take care of it first thing on Monday.

Comment (foxbunny) [reply indico/web/client/styles/base/_utilities.scss]: Done

File: indico/modules/categories/templates/display/event_list.html [hunk 2]
```diff
@@ -59,6 +59,10 @@
     <div class="event-list"
          data-show-future-events-url="{{ url_for('.show_future_events', category) }}"
          data-show-past-events-url="{{ url_for('.show_past_events', category) }}">
+        <h2 id="event-list">
```
Comment (tomasr8) [inline indico/modules/categories/templates/display/event_list.html:62]: Is this id used anywhere? I just see the class name (with the same name)

**Comment by foxbunny** [comment]
Implement via #5945

**Pull request closed without merging**